src/engine.rs

```diff
@@ -100,6 +111,38 @@ mod tests {
         )
         .unwrap();
 
-        assert!(function_run_result.runtime > Duration::from_millis(5));
+        assert!(function_run_result.runtime > RUNTIME_THRESHOLD);
+    }
+
+    #[test]
+    fn test_memory_usage_under_threshold() {
+        let function_run_result = run(
+            Path::new("tests/benchmarks/hello_world.wasm").to_path_buf(),
+            Path::new("tests/benchmarks/hello_world.json").to_path_buf(),
+        )
+        .unwrap();
+
+        assert_eq!(function_run_result.memory_usage, 17);
+    }
+
+    #[test]
+    fn test_memory_usage_over_threshold() {
+        let function_run_result = run(
+            Path::new("tests/benchmarks/hello_42_pages.wasm").to_path_buf(),
+            Path::new("tests/benchmarks/hello_42_pages.json").to_path_buf(),
+        )
+        .unwrap();
+
+        assert_eq!(function_run_result.memory_usage, 42);
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_panic() {
+        run(
+            Path::new("tests/benchmarks/stack_overflow.wasm").to_path_buf(),
+            Path::new("tests/benchmarks/stack_overflow.json").to_path_buf(),
+        )
+        .unwrap();
```

Comment:
It seems that the run function itself doesn't panic, it returns an `Err` result. We should test for that instead of having a panic in the test that we're looking for.

Reply:
Right, it just seems like an extra step that we don't need. We could construct every test by creating an error and unwrapping, but why would we?